be/src/storage/rowset/segment_rewriter.cpp

```diff
@@ -2,18 +2,116 @@
 
 #include "segment_rewriter.h"
 
-namespace starrocks {
+#include "gutil/strings/substitute.h"
+#include "storage/fs/fs_util.h"
+#include "storage/rowset/segment_writer.h"
+#include "storage/vectorized/chunk_helper.h"
+#include "util/crc32c.h"
+#include "util/slice.h"
 
+namespace starrocks {
 namespace segment_v2 {
 
 SegmentRewriter::SegmentRewriter() {}
 
 SegmentRewriter::~SegmentRewriter() {}
 
+Status parse_segment_footer(std::unique_ptr<fs::ReadableBlock>* rblock, SegmentFooterPB* footer,
```

Comment:
similar with https://github.com/StarRocks/starrocks/blob/d78926e4ad897c016cbde18a4a2c394d4d5b84fb/be/src/storage/rowset/segment.cpp#L129-L206, can reuse?